src/main/histFileMod.F90

```diff
@@ -914,68 +932,83 @@ subroutine htapes_fieldlist()
        ! Add the field to the tape if specified via namelist (FINCL[1-max_tapes]),
        ! or if it is on by default and was not excluded via namelist (FEXCL[1-max_tapes]).
 
-       do f = 1,nallhistflds
-          allhistfldname = allhistfldlist(f)%field%name
-          call list_index (fincl(1,t), allhistfldname, ff)
+       ! 8) TODO DONE do f = 1, maxsplitfiles where needed; search "do t"
+       file_loop1: do f = 1, maxsplitfiles
+          do fld = 1, nallhistflds
+             allhistfldname = allhistfldlist(fld)%field%name
+             call list_index (fincl(1,t), allhistfldname, ff)
 
-          if (ff > 0) then
+             if (ff > 0) then
 
-             ! if field is in include list, ff > 0 and htape_addfld
-             ! will be called for field
+                ! if field is in include list, ff > 0 and htape_addfld
+                ! will be called for field
 
-             avgflag = getflag (fincl(ff,t))
-             call htape_addfld (t, f, avgflag)
+                avgflag = getflag (fincl(ff,t))
+                if (f == instantaneous_file_index .and. avgflag == 'I') then
+                   call htape_addfld (t, f, fld, avgflag)
```

Comment:
Would I get the same outcome if I moved "file_loop" just around the three "call htape_addfld" statements?

Reply:
Yes, I think that should work. And it might make it easier to read if you do that. Since, the mainl parts where you need the "f" index is around the htape_addfld calls.

Although you also need an file_loop around the lines in 975 to 986 as well. So you'd end up with three loops rolled out rather than a big loop around the flds loops.

Reply:
> [...]
> Although you also need an file_loop around the lines in 975 to 986 as well. So you'd end up with three loops rolled out rather than a big loop around the flds loops.

Looking at this again I see what you mean, and now that I got all the tests to pass, I'm leaning against moving the file_loop, in case I break things. So i'm resolving this conversation.